test/unit/org/apache/cassandra/db/virtual/PrimaryIdTableTest.java

```diff
@@ -0,0 +1,322 @@
+        ByteBuffer value = ByteBuffer.wrap(new byte[10]);
+        execute("INSERT INTO %s (key, value) VALUES (?, ?)", o1, value);
+        execute("INSERT INTO %s (key, value) VALUES (?, ?)", o2, value);
+        Util.flushTable(KEYSPACE, table);
+
+        ResultSet rs = executeNetWithPaging("SELECT * FROM vts.primary_ids WHERE keyspace_name = ? AND table_name = ? AND token_value = 10001",
+                                            10, KEYSPACE, table);
+        List<Row> all = rs.all();
+        assertEquals(0, all.size());
+        assertEquals(1, scanned.get());
+    }
+
+    @Test
+    public void testPrimaryIdTableDuplicates()
+    {
+        // 0xc25f118f072d6ba5cab7fb1468ace617 hashes to 1563004846366
+        ByteBuffer dup = Murmur3Partitioner.LongToken.keyForToken(1563004846366L);
+        // -19, 68, -61 (0xed44c3) hashes to 1563004846366
+        ByteBuffer dup2 = ByteBuffer.wrap(new byte[]{ -19, 68, -61 });
+        ByteBuffer value = ByteBuffer.wrap(new byte[10]);
+        execute("INSERT INTO %s (key, value) VALUES (?, ?)", dup, value);
+        execute("INSERT INTO %s (key, value) VALUES (?, ?)", dup2, value);
+        Util.flushTable(KEYSPACE, table);
+
+        ResultSet rs = executeNetWithPaging("SELECT * FROM vts.primary_ids WHERE keyspace_name = ? AND table_name = ? AND token_value = 1563004846366",
+                                            10, KEYSPACE, table);
+        List<Row> all = rs.all();
+        assertEquals(2, all.size());
+        assertEquals(BigInteger.valueOf(1563004846366L), all.get(0).get("token_value", BigInteger.class));
+        assertEquals(BigInteger.valueOf(1563004846366L), all.get(1).get("token_value", BigInteger.class));
+        assertEquals("0xc25f118f072d6ba5cab7fb1468ace617", all.get(0).getString("key"));
+        assertEquals("0xed44c3", all.get(1).getString("key"));
+        assertEquals(2, scanned.get());
+    }
+
+    @Test
+    public void testSameKeyInMultipleSSTables()
+    {
+        String table = createTable("CREATE TABLE %s (key blob PRIMARY KEY, value blob)");
+
+        ByteBuffer key = Murmur3Partitioner.LongToken.keyForToken(1);
+        ByteBuffer value = ByteBuffer.wrap(new byte[10]);
+        execute("INSERT INTO %s (key, value) VALUES (?, ?)", key, value);
+        Util.flushTable(KEYSPACE, table);
+        value = ByteBuffer.wrap(new byte[100]);
+        execute("INSERT INTO %s (key, value) VALUES (?, ?)", key, value);
+        Util.flushTable(KEYSPACE, table);
+
+        ResultSet rs = executeNetWithPaging("SELECT * FROM vts.primary_ids WHERE keyspace_name = ? AND table_name = ?",
+                                            10, KEYSPACE, table);
+        List<Row> all = rs.all();
+        assertEquals(1, all.size());
+        Row row = all.get(0);
+        assertEquals(BigInteger.valueOf(1), row.get("token_value", BigInteger.class));
+        long size = row.get("size_estimate", Long.class);
+        assertTrue(size >= 110 && size < 200);
```

Comment:
I'm curious about this particular range for the estimate check. From my tests, size is always 162. Should we just check for `size == 126` with a nice comment onto why that is happening?

Reply:
Got it, that makes sense. Then maybe adding just a quick comment stating that for curious people like me will do :-P